devdoc/LDAPSecurityManager.md

```diff
@@ -0,0 +1,92 @@
+# LDAP Security Manager Init
+This document covers the main steps taken by the LDAP Security Manager during initialization. It is important to
+note that the LDAP Security Manager uses the LDAP protocol to access an Active Directory, AD. The AD is the store for
+users, groups, permissions, resources, and more. The term LDAP is generally overused to refer to both.
+
+## LDAP Instances ##
+Each service and/or component using the LDAP security manager gets its own instance of the security manager. This 
+includes a unique connection pool (see below). All operations described apply to each LDAP instance.
+
+## Initialization Steps
+The following sections cover the main steps taken during initialization
+
+### Load Configuration
+The following items are loaded from the configuration
+
+#### AD Hosts
+The LDAP Security Manager supports using multiple ADs. The FQDN or IP address of each AD host is read from configuration
+data and stored internally. The source is a comma separated list stored in the _ldapAddress_ config value. Each entry
+is added to a pool of ADs.
+
+**Note that all ADs are expected to use the same credentials and have the same configuration**
+
+#### AD Credentials
+AD credentials consist of a username and password. The LDAP security manager users these to perform all operations
+on behalf of users and components in the cluster. There are three potential sources for credentials.
+
+1. A Kubernetes secret. If the _ldapAdminSecretKey_ config value is set, but _ldapAdminVaultId_ is not (see 2) then
+the AD credentials are retrieved as Kubernetes secrets.
+2. If both _ldapAdminSecretKey_ and _ldapAdminVaultId_ config values are present, the AD credentials are retrieved
+from the vault.
+3. Hardcoded values from the _systemCommonName_ and _systemPassword_ config values stored in the environment.xml file.
+
```

Comment:
Should we warn that #3 is not a secure method of storage?

Reply:
It's not really the purpose of the document. I think that is something reserved for end user docs discussing how to setup a secure system.

Reply:
fair enough